src/Symfony/Component/Console/CommandLoader/ContainerCommandLoader.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Symfony\Component\Console\CommandLoader;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Console\Exception\CommandNotFoundException;
+
+/**
+ * Loads commands from a PSR-11 container.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+class ContainerCommandLoader implements CommandLoaderInterface
+{
+    private $container;
+    private $commandMap;
+
+    /**
+     * @param ContainerInterface $container  A container from which to load command services
+     * @param array              $commandMap An array with command names as keys and service ids as values
+     */
+    public function __construct(ContainerInterface $container, array $commandMap)
```

Comment:
if container can be keyed by command names, then we won't need $commandMap (we'll need "$commandNames"), which would be enough